Library/Homebrew/livecheck/strategy/github_latest.rb

```diff
@@ -73,32 +71,80 @@ def self.generate_input_values(url)
+          values[:username] = match[:username]
+          values[:repository] = match[:repository]
 
           values
         end
 
-        # Generates a URL and regex (if one isn't provided) and passes them
-        # to {PageMatch.find_versions} to identify versions in the content.
+        # Uses a regex to match the version from release JSON or, if a block is
+        # provided, passes the JSON to the block to handle matching. With
+        # either approach, an array of unique matches is returned.
+        #
+        # @param content [Array] list of releases
+        # @param regex [Regexp] a regex used for matching versions in the content
+        # @param block [Proc, nil] a block to match the content
+        # @return [Array]
+        sig {
+          params(
+            content: T::Array[T::Hash[String, T.untyped]],
+            regex:   Regexp,
+            block:   T.nilable(Proc),
+          ).returns(T::Array[String])
+        }
+        def self.versions_from_content(content, regex, &block)
+          if block.present?
+            block_return_value = if regex.present?
+              yield(content, regex)
+            else
+              yield(content)
+            end
+            return Strategy.handle_block_return(block_return_value)
+          end
+
+          content.reject(&:blank?).map do |release|
+            next if release["draft"] || release["prerelease"]
+
+            value = T.let(nil, T.untyped)
+            VERSION_KEYS.find do |key|
+              value = release[key]&.match(regex)&.to_s
+            end
+            value
+          end.compact.uniq
+        end
+
+        # Generates the GitHub API URL for the repository's "latest" release
+        # and identifies the version from the JSON response.
         #
         # @param url [String] the URL of the content to check
         # @param regex [Regexp] a regex used for matching versions in content
         # @return [Hash]
         sig {
           params(
-            url:    String,
-            regex:  T.nilable(Regexp),
-            unused: T.nilable(T::Hash[Symbol, T.untyped]),
-            block:  T.nilable(Proc),
+            url:     String,
+            regex:   Regexp,
+            _unused: T.nilable(T::Hash[Symbol, T.untyped]),
+            block:   T.nilable(Proc),
           ).returns(T::Hash[Symbol, T.untyped])
         }
-        def self.find_versions(url:, regex: nil, **unused, &block)
+        def self.find_versions(url:, regex: DEFAULT_REGEX, **_unused, &block)
+          match_data = { matches: {}, regex: regex, url: url }
+
           generated = generate_input_values(url)
+          return match_data if generated.blank?
+
+          match_data[:url] = generated[:url]
+
+          release = GitHub.get_latest_release(generated[:username], generated[:repository])
+          versions_from_content([release], regex, &block).each do |match_text|
```

Comment:
```suggestion
          versions_from_content(release, regex, &block).each do |match_text|
```

Based on the `GithubReleases` PR, I know that `#versions_from_content` is accepting an array because you aim to use `GithubReleases#versions_from_content` in `GithubLatest`. I don't see this as a viable approach based on the current `GithubReleases#versions_from_content` because a `GithubLatest` strategy block would be passing in an array as the argument instead of one JSON object.

I don't think anyone is going to expect that the argument will be an array with one JSON object and this could trip people up. This would also require every `GithubLatest` `strategy` block to do something like `releases.first`, which complicates the boilerplate code for the strategy blocks.

If both strategies share the same `#versions_from_content` method, then we're either left with this unsatisfactory situation or we would have to adapt `GithubReleases#versions_from_content` to properly handle either a JSON object or an array of JSON objects.

From my perspective, it's easier to just keep these `#versions_from_content` methods separate. There are situations that apply to `GithubReleases` that don't apply to `GithubLatest` (e.g., prerelease and draft releases can't be "latest", so we don't need to guard against them in `GithubLatest`).

Regardless of whether we have two separate `#versions_from_content` methods or adapt the `GithubRelease` method to also work with a JSON object, `GithubLatest` should be passing in a JSON object. For the purposes of this PR, it allows us to simplify the existing `#versions_from_content` method. We already have to modify the logic of that method to fix a bug, so we may as well simplify at the same time.